src/dataflow/decode/csv.rs

```diff
@@ -42,32 +50,86 @@ where
                     // This is mainly an aesthetic/performance-golfing
                     // issue as I doubt it will ever be a bottleneck.
                     for (line, line_no) in &*lines {
-                        let mut csv_reader = csv::ReaderBuilder::new()
-                            .has_headers(false)
-                            .delimiter(delimiter)
-                            .from_reader(line.as_slice());
-                        for result in csv_reader.records() {
-                            let record = result.unwrap();
-                            if record.len() != n_cols {
-                                events_error += 1;
-                                error!(
-                                    "CSV error: expected {} columns, got {}. Ignoring row.",
-                                    n_cols,
-                                    record.len()
-                                );
-                                continue;
+                        // We only want to process utf8 strings, as this ensures that all fields
+                        // will be utf8 as well, allowing some unsafe shenanigans.
+                        if std::str::from_utf8(line.as_slice()).is_err() {
+                            events_error += 1;
+                            error!("CSV error: input text is not utf8");
+                        } else {
+                            // Reset the reader to read a new series of records.
+                            csv_reader.reset();
+                            if let Some(line_no) = line_no {
+                                csv_reader.set_line(*line_no as u64);
+                            }
+
+                            let mut input = line.as_slice();
+                            let mut buffer_valid = 0;
+                            let mut bounds_valid = 0;
+                            let mut done = false;
+
+                            while !done {
+
+                                // Note that we protect the first element of `bounds`, a zero, so that ranges are easier to extract below.
+                                let (result, in_read, _out_wrote, ends_wrote) = csv_reader
```

Comment:
`_out_wrote` should not have a leading underscore

Reply:
Good point, thanks!